chore: Reduce enzymes usage 1/3

timi-codes opened the pull request:
## **Description**

Convert the list of unit tests to use react-testing-library instead of enzyme for the following files:
```
'app/components/Views/Root/index.test.tsx',
'app/components/Views/NavigationUnitTest/TestScreen3.test.js',
'app/components/Views/NavigationUnitTest/TestScreen2.test.js',
'app/components/Views/NavigationUnitTest/TestScreen1.test.js',
'app/components/Views/ImportPrivateKeySuccess/index.test.tsx',
'app/components/Views/AccountBackupStep1B/index.test.tsx',
'app/components/UI/WebviewProgressBar/index.test.tsx',
'app/components/UI/SeedphraseModal/index.test.tsx',
'app/components/UI/PhishingModal/index.test.tsx',
'app/components/UI/FoxScreen/index.test.tsx',
'app/components/UI/FadeView/index.test.tsx',
'app/components/UI/CustomAlert/index.test.tsx',
'app/components/UI/Confetti/index.test.tsx',
'app/components/UI/Collectibles/index.test.tsx',
'app/components/UI/Button/index.test.tsx',
'app/components/UI/AnimatedSpinner/index.test.tsx',
'app/components/UI/AddCustomToken/index.test.tsx',
```

## **Related issues**

Fixes: #8616 

## **Manual testing steps**

Not Applicable

## **Screenshots/Recordings**

Not Applicable

## **Pre-merge author checklist**

- [x] I’ve followed [MetaMask Coding Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've clearly explained what problem this PR is solving and how it is solved.
- [x] I've linked related issues
- [ ] I've included manual testing steps
- [ ] I've included screenshots/recordings if applicable
- [ ] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [ ] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.
- [x] I’ve properly set the pull request status:
  - [ ] In case it's not yet "ready for review", I've set it to "draft".
  - [x] In case it's "ready for review", I've changed it from "draft" to "non-draft".

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

timi-codes committed:
fix(#8616): Reduce enzyme usage in unit test by 25% - [1/3]

timi-codes committed:
fix(#8616): Reduce enzyme usage in unit test by 25% - [1/3]

Cal-L reviewed (commented):
Left comments of simplified tests using existing utility functions

Cal-L commented:
@timi-codes Left some comments

timi-codes commented:
Thanks for the feedback @Cal-L 

I have updated 15/17 of the test files and made the recommended changes. The remaining two files are `app/components/Views/Root/index.test.tsx`  and `app/components/UI/Confetti/index.test.tsx`.

The root file snapshot render null and the confetti snapshot change every time because of the randomness of the Views. 
Do you have any suggestion on how to fix this?

<img width="676" alt="image" src="https://github.com/MetaMask/metamask-mobile/assets/7896429/2c0ebb25-1dae-41dc-a7e5-649ba609e4a6">
<img width="429" alt="image" src="https://github.com/MetaMask/metamask-mobile/assets/7896429/077d4509-b43c-424b-b919-68e7caac563a">

Cal-L commented:
> Thanks for the feedback @Cal-L
> 
> I have updated 15/17 of the test files and made the recommended changes. The remaining two files are `app/components/Views/Root/index.test.tsx` and `app/components/UI/Confetti/index.test.tsx`.
> 
> The root file snapshot render null and the confetti snapshot change every time because of the randomness of the Views. Do you have any suggestion on how to fix this?
> 
> <img alt="image" width="676" src="https://private-user-images.githubusercontent.com/7896429/309115199-2c0ebb25-1dae-41dc-a7e5-649ba609e4a6.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDkzMTc4OTQsIm5iZiI6MTcwOTMxNzU5NCwicGF0aCI6Ii83ODk2NDI5LzMwOTExNTE5OS0yYzBlYmIyNS0xZGFlLTQxZGMtYTdlNS02NDliYTYwOWU0YTYucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDMwMSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDAzMDFUMTgyNjM0WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9OTIyN2M5M2Y1NGU3MDcwMDJjZTQwMzNkMjU2OTJlZjM5OTY0YjAxZTkyOWM0NjBiYTAxMTUyODk2MWJlMDIyYSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.urf-EOrt1PMqW4GYIbGI6w1g9RZ81i1s2BF5j7RTk2M"> <img alt="image" width="429" src="https://private-user-images.githubusercontent.com/7896429/309115159-077d4509-b43c-424b-b919-68e7caac563a.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDkzMTc4OTQsIm5iZiI6MTcwOTMxNzU5NCwicGF0aCI6Ii83ODk2NDI5LzMwOTExNTE1OS0wNzdkNDUwOS1iNDNjLTQyNGItYjkxOS02OGU3Y2FhYzU2M2EucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDMwMSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDAzMDFUMTgyNjM0WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9ZmM0M2NmZTc4ZmJjZWVhYTE3MmUyM2EzNjNmZjJiNTdmZTc3M2YyY2YyYjcyNzFlMzdhZWZkM2I5ZTg4OTM3OSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.i0DqFpp6KPD6x7rRNOHXQpOUJmabIJduoOpbDVi98aU">

On `Confetti.test.tsx`, since the components being tested have non deterministic styles, I'd recommend just mocking `react-native-confetti` and `react-native-confetti-cannon` for now:

```
jest.mock('react-native-confetti');
jest.mock('react-native-confetti-cannon');
```

The snapshot will reduce to only return null but that's fine for now since this isn't a critical component.

Cal-L commented:
> Thanks for the feedback @Cal-L
> 
> I have updated 15/17 of the test files and made the recommended changes. The remaining two files are `app/components/Views/Root/index.test.tsx` and `app/components/UI/Confetti/index.test.tsx`.
> 
> The root file snapshot render null and the confetti snapshot change every time because of the randomness of the Views. Do you have any suggestion on how to fix this?
> 
> <img alt="image" width="676" src="https://private-user-images.githubusercontent.com/7896429/309115199-2c0ebb25-1dae-41dc-a7e5-649ba609e4a6.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDkzMTc4OTQsIm5iZiI6MTcwOTMxNzU5NCwicGF0aCI6Ii83ODk2NDI5LzMwOTExNTE5OS0yYzBlYmIyNS0xZGFlLTQxZGMtYTdlNS02NDliYTYwOWU0YTYucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDMwMSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDAzMDFUMTgyNjM0WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9OTIyN2M5M2Y1NGU3MDcwMDJjZTQwMzNkMjU2OTJlZjM5OTY0YjAxZTkyOWM0NjBiYTAxMTUyODk2MWJlMDIyYSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.urf-EOrt1PMqW4GYIbGI6w1g9RZ81i1s2BF5j7RTk2M"> <img alt="image" width="429" src="https://private-user-images.githubusercontent.com/7896429/309115159-077d4509-b43c-424b-b919-68e7caac563a.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDkzMTc4OTQsIm5iZiI6MTcwOTMxNzU5NCwicGF0aCI6Ii83ODk2NDI5LzMwOTExNTE1OS0wNzdkNDUwOS1iNDNjLTQyNGItYjkxOS02OGU3Y2FhYzU2M2EucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDMwMSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDAzMDFUMTgyNjM0WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9ZmM0M2NmZTc4ZmJjZWVhYTE3MmUyM2EzNjNmZjJiNTdmZTc3M2YyY2YyYjcyNzFlMzdhZWZkM2I5ZTg4OTM3OSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.i0DqFpp6KPD6x7rRNOHXQpOUJmabIJduoOpbDVi98aU">

For the Root component snapshot tests, you can update the test file to look like this:

<img width="554" alt="image" src="https://github.com/MetaMask/metamask-mobile/assets/10508597/017cc581-9eff-4312-9b3e-78a2fb7c056d">

After that, you can remove some of the mocks

<img width="510" alt="image" src="https://github.com/MetaMask/metamask-mobile/assets/10508597/25a33662-2a4b-4edd-bcb5-0f959625e74d">

Your Root snapshot should update and the output should remain deterministic

Cal-L commented:
@timi-codes Left some comments regarding Confetti and Root components

timi-codes commented:
> > Thanks for the feedback @Cal-L
> > I have updated 15/17 of the test files and made the recommended changes. The remaining two files are `app/components/Views/Root/index.test.tsx` and `app/components/UI/Confetti/index.test.tsx`.
> > The root file snapshot render null and the confetti snapshot change every time because of the randomness of the Views. Do you have any suggestion on how to fix this?
> > <img alt="image" width="676" src="https://private-user-images.githubusercontent.com/7896429/309115199-2c0ebb25-1dae-41dc-a7e5-649ba609e4a6.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDkzMTc4OTQsIm5iZiI6MTcwOTMxNzU5NCwicGF0aCI6Ii83ODk2NDI5LzMwOTExNTE5OS0yYzBlYmIyNS0xZGFlLTQxZGMtYTdlNS02NDliYTYwOWU0YTYucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDMwMSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDAzMDFUMTgyNjM0WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9OTIyN2M5M2Y1NGU3MDcwMDJjZTQwMzNkMjU2OTJlZjM5OTY0YjAxZTkyOWM0NjBiYTAxMTUyODk2MWJlMDIyYSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.urf-EOrt1PMqW4GYIbGI6w1g9RZ81i1s2BF5j7RTk2M"> <img alt="image" width="429" src="https://private-user-images.githubusercontent.com/7896429/309115159-077d4509-b43c-424b-b919-68e7caac563a.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDkzMTc4OTQsIm5iZiI6MTcwOTMxNzU5NCwicGF0aCI6Ii83ODk2NDI5LzMwOTExNTE1OS0wNzdkNDUwOS1iNDNjLTQyNGItYjkxOS02OGU3Y2FhYzU2M2EucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDMwMSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDAzMDFUMTgyNjM0WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9ZmM0M2NmZTc4ZmJjZWVhYTE3MmUyM2EzNjNmZjJiNTdmZTc3M2YyY2YyYjcyNzFlMzdhZWZkM2I5ZTg4OTM3OSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.i0DqFpp6KPD6x7rRNOHXQpOUJmabIJduoOpbDVi98aU">
> 
> For the Root component snapshot tests, you can update the test file to look like this:
> 
> <img alt="image" width="554" src="https://private-user-images.githubusercontent.com/10508597/309658975-017cc581-9eff-4312-9b3e-78a2fb7c056d.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDk1OTkyMzQsIm5iZiI6MTcwOTU5ODkzNCwicGF0aCI6Ii8xMDUwODU5Ny8zMDk2NTg5NzUtMDE3Y2M1ODEtOWVmZi00MzEyLTliM2UtNzhhMmZiN2MwNTZkLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAzMDUlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMzA1VDAwMzUzNFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPThmMWJhMThjMTQwMWRmMjUyMDQwNTNiN2EwY2Q2MjhhMmZhZjJmMTlhMDQ4ZTcwNzVmNDIyOTcyMGI4ZjJjM2YmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.-yZNuERRkB81uboM7pUBNV3ZBYyCUCAQCoEfOwjZGVk">
> After that, you can remove some of the mocks
> 
> <img alt="image" width="510" src="https://private-user-images.githubusercontent.com/10508597/309659173-25a33662-2a4b-4edd-bcb5-0f959625e74d.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDk1OTkyMzQsIm5iZiI6MTcwOTU5ODkzNCwicGF0aCI6Ii8xMDUwODU5Ny8zMDk2NTkxNzMtMjVhMzM2NjItMmE0Yi00ZWRkLWJjYjUtMGY5NTk2MjVlNzRkLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAzMDUlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMzA1VDAwMzUzNFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWViODZhY2NlZDkyOWIwMmEyMWI3NTZmMzI5OWY1Y2FjMDkyMTE3MDA2NTEwYjRmYTM4MDUwNzc2NjJjZWNkMmEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.nDDwbJjJFGF8EGAKJAL7wtuG0rm0OdGwA9Zt3HmzDDc">
> Your Root snapshot should update and the output should remain deterministic

@Cal-L Thank you. 

For the Root file, I was referring to `app/components/Views/Root/index.test.tsx` not `app/components/Views/confirmations/components/SignatureRequest/Root/Root.test.tsx`. However, I already fixed it by mocking `configureStore`. 

All the files are now ready to be reviewed.

Cal-L commented on app/components/UI/Button/index.test.tsx:
```diff
@@ -1,10 +1,10 @@
 describe('Button', () => {
   it('should render correctly', () => {
-    const wrapper = shallow(<Button />);
+    const wrapper = render(<Button />);
```
check for isJSON

Cal-L commented on app/components/UI/FadeView/index.test.tsx:
```diff
@@ -1,10 +1,10 @@
 describe('FadeView', () => {
   it('should render correctly', () => {
-    const wrapper = shallow(<FadeView visible />);
+    const wrapper = render(<FadeView visible />);
```
check for isJSON

Cal-L reviewed (commented):
Left some comments @timi-codes

timi-codes committed:
Merge branch 'main' into reduce-enzymes-usage-1/3

timi-codes committed:
Merge branch 'main' into reduce-enzymes-usage-1/3

Cal-L reviewed (approved):
LGTM, thanks for your contribution!

timi-codes committed:
Merge branch 'main' into reduce-enzymes-usage-1/3

timi-codes commented:
Thank you @Cal-L . The 2/3 is also ready for review https://github.com/MetaMask/metamask-mobile/pull/8823

Cal-L committed:
Merge branch 'main' into reduce-enzymes-usage-1/3

Cal-L committed:
Merge branch 'main' into reduce-enzymes-usage-1/3

Cal-L commented:
@timi-codes  Merging into a branch based on main and will follow up with a PR to merge into main.
